### DISCUSSION TITLE
How do I get current resolution and list of supported resolutions of a monitor?

### DISCUSSION THREAD

**Discussion by ayonization** [oshi/oshi#2658] [Q&A] [answered]
I noticed that the Xrandr class has just one method that only returns the edid string from the entire xrandr verbose output.
As far as I know, the EDID will not contain current resolutinon information for the monitor, while the xrandr command does spit out that information.

Also, is there a way to expose the capability to change the resolutions for a particular display, given that the resolution is supported by the monitor, I know this can be done using xrandr, but it would be super helpful if it was possible through OSHI.

**Comment by dbwiddis** [comment, marked as answer]
> I noticed that the Xrandr class has just one method that only returns the edid string from the entire xrandr verbose output. As far as I know, the EDID will not contain current resolutinon information for the monitor, while the xrandr command does spit out that information.

Correct, OSHI's display class is pretty rudimentary, just counting displays and giving the edid.

If you are able to get some of that additional information cross-platform, I'd welcome a PR improving on the `Display` class.
 - macOS has command-line `system_profiler SPDisplaysDataType`
 - Windows has `EnumDisplaySettings`

> Also, is there a way to expose the capability to change ...

OSHI only reports information, and has avoided making changes like this.

See related request for a "kill process" feature here: https://github.com/oshi/oshi/issues/277; the general philosophy remains the same.
